main_test.go

```diff
@@ -77,7 +77,18 @@ func TestLifecycle(t *testing.T) {
 		{
 			name: "url denied",
 			inlineRules: `
-			SecRuleEngine On\nSecRule REQUEST_URI \"@streq /hello?name=panda\" \"id:101,phase:1,t:lowercase,deny\"
+			SecRuleEngine On\nSecRule REQUEST_URI \"@beginsWith /hello?name=panda\" \"id:101,phase:1,t:lowercase,deny\"
+			`,
+			requestHdrsAction:  types.ActionPause,
+			requestBodyAction:  types.ActionContinue,
+			responseHdrsAction: types.ActionContinue,
+			responded403:       true,
+			respondedNullBody:  false,
+		},
+		{
+			name: "XSS query argument denied",
+			inlineRules: `
+			SecRuleEngine On\nInclude @owasp_crs\/REQUEST-941-APPLICATION-ATTACK-XSS.conf
 			`,
 			requestHdrsAction:  types.ActionPause,
```

Comment:
If we don't set Pause here, then we fail because local response is empty.

Reply:
Adding the Self-contained config rules that lead to triggering the deny status if one rule is triggered, I'm experiencing a 403 local response. I'm using this line: `SecRuleEngine On\nSecDefaultAction \"phase:1,log,auditlog,deny,status:403\"\nSecDefaultAction \"phase:2,log,auditlog,deny,status:403\"\nInclude @owasp_crs\/REQUEST-941-APPLICATION-ATTACK-XSS.conf`